persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/ProxyProtocolQueryInterceptor.java

```diff
@@ -50,91 +126,152 @@ public void initialize() {}
-                                  UUID.nameUUIDFromBytes(publicAddress.getAddress())));
-                      break;
-                    case "tokens":
-                      // All entries handle the entire token ring. This prevents some driver from
-                      // crashing when `tokens` is null.
-                      rows.result
-                          .rows
-                          .get(0)
-                          .set(
-                              index,
-                              SetType.getInstance(UTF8Type.instance, false)
-                                  .decompose(
-                                      Collections.singleton(
-                                          DatabaseDescriptor.getPartitioner()
-                                              .getMinimumToken()
-                                              .toString())));
-                  }
-                  ++index;
-                }
-              }
+    ResultSet resultSet = new ResultSet(selectStatement.getResultMetadata(), rows);
+    return Single.just(new ResultMessage.Rows(resultSet));
+  }
+
+  @Override
+  public void register(EventListener listener) {
+    listeners.add(listener);
+  }
+
+  private void resolvePeers() {
+    if (!Strings.isNullOrEmpty(proxyDnsName)) {
+      try {
+        Set<InetAddress> resolved = resolver.resolve(proxyDnsName);
+
+        if (!peers.equals(resolved)) {
+          // Generate listener events based on the differences between this and the previous
+          // resolved peers.
+          Sets.SetView<InetAddress> added = Sets.difference(resolved, peers);
+          Sets.SetView<InetAddress> removed = Sets.difference(peers, resolved);
+
+          for (EventListener listener : listeners) {
+            for (InetAddress peer : added) {
+              listener.onJoinCluster(peer, proxyPort);
+              listener.onUp(peer, proxyPort);
+            }
+            for (InetAddress peer : removed) {
+              listener.onLeaveCluster(peer, proxyPort);
             }
-            return result;
-          });
+          }
+          peers = resolved;
+        }
+      } catch (UnknownHostException e) {
+        throw new ServerError("Unable to resolve DNS for proxy protocol peers table", e);
+      }
+      scheduler.schedule(this::resolvePeers, resolveDelaySecs, TimeUnit.SECONDS);
     }
   }
 
-  @Override
-  public void register(IEndpointLifecycleSubscriber subscriber) {
-    // TODO: Monitor A-records for DNS to generate events for stargate peers
+  /**
+   * Creates {@link ByteBuffer} value for a given column name. Unhandled names and aggregates return
+   * a null value.
+   *
+   * @param name
+   * @param publicAddress
+   * @return A {@link ByteBuffer} value for a given system local/peers column.
+   */
+  private static ByteBuffer buildColumnValue(String name, InetAddress publicAddress) {
+    switch (name) {
+      case "key":
+        return UTF8Type.instance.decompose("local");
```

Comment:
nit: `"local"` extract to constant

Reply:
What's the benefit of extracting a constant that's used one time?